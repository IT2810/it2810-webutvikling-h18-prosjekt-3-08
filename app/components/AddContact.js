import React from 'react';
import {StyleSheet, Text, View, KeyboardAvoidingView, TextInput, Button, ScrollView } from 'react-native';


export default class AddContacat extends React.Component {

    constructor() {
        super()
        this.state = {
            firstName: '',
            lastName: '',
            email: '',
            phoneNo: ''
        }
    }

    addAndNavigate() {
        
        let newContact = {
            firstName: this.state.firstName,
            lastName: this.state.lastName,
            email: this.state.email,
            phoneNo: this.state.phoneNo
        }
        this.props.navigation.state.params.addItem(newContact)
        this.props.navigation.navigate('Contacts')
    }

 
    render() {
        let firstNameStyle = this.state.firstName === '' ? styles.textInput : styles.textInput2
        let lasstNameStyle = this.state.lastName === '' ? styles.textInput : styles.textInput2
        let emailStyle = this.state.email === '' ? styles.textInput : styles.textInput2
        let phoneNoStyle = this.state.phoneNo === '' ? styles.textInput : styles.textInput2
        return (
        
            <ScrollView>
                <View style={styles.container}>
                    <KeyboardAvoidingView behavior= "padding" styles= {{flex: 1}} >
                        <TextInput
                            style={firstNameStyle}
                            value={this.state.firstName}
                            placeholder="Firstname"
                            onChangeText={(value) => this.setState({firstName: value})}
                            returnKeyType="go"
                        />
                    </KeyboardAvoidingView>
                    <KeyboardAvoidingView behavior= "padding" styles= {{flex: 1}} >
                        <TextInput
                            style={lasstNameStyle}
                            value={this.state.lastName}
                            placeholder="Lastname"
                            onChangeText={(value) => this.setState({lastName: value})}
                            returnKeyType="go"
                        />
                    </KeyboardAvoidingView>
                    <KeyboardAvoidingView behavior= "padding" styles= {{flex: 1}} >
                        <TextInput
                            style={emailStyle}
                            value={this.state.email}
                            placeholder="Email"
                            onChangeText={(value) => this.setState({email: value})}
                            returnKeyType="go"
                        />
                    </KeyboardAvoidingView>
                    <KeyboardAvoidingView behavior= "padding" styles= {{flex: 1}} >
                        <TextInput
                            style={phoneNoStyle}
                            value={this.state.phoneNo}
                            placeholder="Phone number"
                            onChangeText={(value) => this.setState({phoneNo: value})}
                            returnKeyType="go"
                        />
                    </KeyboardAvoidingView>
                </View>
                <View style={styles.buttonContainer}>
                    <Button
                        title="Add Contact"
                        onPress={this.addAndNavigate.bind(this)}
                        style={{margin: 40}}
                    />
                </View>
            
            </ScrollView>
        )
   }
}

const styles = StyleSheet.create({
   container: {
      backgroundColor: '#fff',
      flex: 1,
      marginTop: 60
   },
   buttonContainer: {
       backgroundColor: '#fff',
       flex: 1,
       marginTop: 40
   },
   textInput: {
    margin: 15,
    marginBottom: 5,
    height: 40,
    color: 'black',
    padding: 5,
    backgroundColor: '#fff',
    borderBottomWidth: 2,
    borderBottomColor: '#ededed',
    },
    textInput2: {
        margin: 20,
        marginBottom: 0,
        height: 40,
        color: 'black',
        padding: 5,
        backgroundColor: '#fff',
        borderBottomWidth: 2,
        borderBottomColor: '#A9E2F3',
    },
})
