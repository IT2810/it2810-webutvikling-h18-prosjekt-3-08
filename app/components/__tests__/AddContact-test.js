import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import AddContact from '../AddContact';


test("Renders correctly", async () => {
    const tree = renderer.create(<AddContact/>).toJSON();
    expect(tree).toMatchSnapshot();
});
