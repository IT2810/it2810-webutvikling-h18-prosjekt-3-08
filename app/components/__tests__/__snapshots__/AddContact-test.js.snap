// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<View>
  <View
    enabled={true}
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "paddingBottom": 0,
        },
      ]
    }
    styles={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Firstname"
      returnKeyType="go"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopColor": "#ededed",
          "borderTopWidth": 2,
          "color": "black",
          "height": 40,
          "margin": 20,
          "marginBottom": 0,
          "padding": 5,
        }
      }
      value=""
    />
  </View>
  <View
    enabled={true}
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "paddingBottom": 0,
        },
      ]
    }
    styles={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Lastname"
      returnKeyType="go"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopColor": "#ededed",
          "borderTopWidth": 2,
          "color": "black",
          "height": 40,
          "margin": 20,
          "marginBottom": 0,
          "padding": 5,
        }
      }
      value=""
    />
  </View>
  <View
    enabled={true}
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "paddingBottom": 0,
        },
      ]
    }
    styles={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email"
      returnKeyType="go"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopColor": "#ededed",
          "borderTopWidth": 2,
          "color": "black",
          "height": 40,
          "margin": 20,
          "marginBottom": 0,
          "padding": 5,
        }
      }
      value=""
    />
  </View>
  <View
    enabled={true}
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "paddingBottom": 0,
        },
      ]
    }
    styles={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Phone number"
      returnKeyType="go"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopColor": "#ededed",
          "borderTopWidth": 2,
          "color": "black",
          "height": 40,
          "margin": 20,
          "marginBottom": 0,
          "padding": 5,
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add Contact
      </Text>
    </View>
  </View>
</View>
`;
